In [7]:
import numpy as np
from statsmodels.stats.weightstats import ztest

altura_turma = np.array([165, 157, 163, 170, 167, 161, 160, 170, 157, 160, 157, 168, 162, 170, 167, 165, 166, 158, 174, 169, 163, 174, 162, 170, 155, 154, 158])
media_nacional = 161

z_statistic, p_value = ztest(altura_turma, value=media_nacional)
print("Teste z para uma amostra:")
print(f"z-statistic: {z_statistic:.4f} \np-value: {p_value:.4f}")

Teste z para uma amostra:
z-statistic: 2.5116 
p-value: 0.0120


**H0​:** A altura média da turma on35 é igual a altura média nacional.

**H1:** A altura média da turma on35 é diferente da altura média nacional.

**Conclusão:**

O p_value 0.0120, sendo menor que o nível de significância 0.05, sugere que a altura média da turma é significativamente diferente de 161 cm e, portanto, **<u>rejeita-se a hipótese nula (H0)</u>**. Logo, há evidência estatística para afirmar que a altura média da turma é significativamente diferente da altura média nacional feminina de 161 cm.